src/deployer.ts

```diff
@@ -76,7 +80,7 @@ export abstract class Deployer extends EventEmitter {
   /**
    * Deploy the app.
    */
-  public abstract deploy(): Promise<void>;
+  public abstract deploy<R extends DeployerResult>(): Promise<void | R>;
```

Comment:
let the deploy function return a new type that extends from DeployerResult

Reply:
I suppose it's not possible to get rid of `void` here without introducing a breaking change?

Reply:
That would be a breaking change, but we are still in beta :)

Reply:
True... but it feels like it'd be more trouble than it's worth. plugin-deploy-retrieve and plugin-functions are the only plugins that use the `Deployer` interface, but we'd still have to go a bump sf-plugins-core in every other plugin